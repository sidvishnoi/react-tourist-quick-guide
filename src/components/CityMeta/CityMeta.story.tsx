import { action } from '@storybook/addon-actions';
import { number, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import CityMeta from '.';

const stories = storiesOf('CityMeta', module);
stories.addDecorator(withKnobs);

stories.add('CityMeta', () => {
  return (
    <CityMeta
      onMoveButtonClick={() => action('onMoveButtonClick')()}
      onRemoveButtonClick={() => action('onRemoveButtonClick')()}
      name={text('name', 'City Name')}
      distance={number('distance', 40)}
    />
  );
});
