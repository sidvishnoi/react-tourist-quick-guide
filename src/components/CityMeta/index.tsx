import * as React from 'react';
import styled from 'styled-components';

import IconButton from '../IconButton';

const CityMeta = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  background: #000;
  color: #fff;
  width: 200px;
  padding: 0.2em;

  .buttons {
    display: flex;
    justify-content: space-between;

    @media print {
      display: none;
    }
  }

  @media print {
    justify-content: flex-start;
    width: auto;
    max-width: 200px;
    background: #fff;
    color: #000;
  }
`;

const CityName = styled.h2`
  font-size: 2rem;
  text-align: center;
  margin: 0.5rem auto;
  padding: 0.2rem 0.5rem;

  @media print {
    padding: 0;
    margin: 0;
    text-align: left;
  }
`;

const Distance = styled.div`
  text-align: right;
  padding: 0.5rem;

  @media print {
    padding: 0;
    text-align: left;
  }
`;

export interface CityMetaProps {
  name: string;
  onMoveButtonClick: (name: string) => void;
  onRemoveButtonClick: (name: string) => void;
  distance: number;
}

export default (props: CityMetaProps) => {
  const { name, onMoveButtonClick, onRemoveButtonClick, distance } = props;
  return (
    <CityMeta>
      <div className="buttons">
        <IconButton
          icon="⬆"
          className="move"
          title="Move destination up in list"
          onClick={() => onMoveButtonClick(name)}
          color="#fff"
          altColor="cornflowerblue"
        />
        <IconButton
          icon="✗"
          className="remove"
          title="Remove from list"
          onClick={() => onRemoveButtonClick(name)}
          color="#fff"
          altColor="red"
        />
      </div>
      <CityName>{name}</CityName>
      <Distance>{distance !== null ? `${distance} KM` : '...'}</Distance>
    </CityMeta>
  );
};
