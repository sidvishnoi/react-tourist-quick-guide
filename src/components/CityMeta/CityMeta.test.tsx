import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import React = require('react');
import * as renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

import CityMeta from '.';

describe('Componets - CityMeta', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <CityMeta
        onMoveButtonClick={() => null}
        onRemoveButtonClick={() => null}
        name="PASS"
        distance={10}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('calls destroyer on remove button click', () => {
    const onMoveButtonClick = (): null => null;
    const onRemoveButtonClick = jest.fn();

    const output = mount(
      <CityMeta
        onMoveButtonClick={onMoveButtonClick}
        onRemoveButtonClick={onRemoveButtonClick}
        name="PASS"
        distance={10}
      />,
    );
    const removeButton = output.find('button.remove');
    expect(onRemoveButtonClick).not.toHaveBeenCalled();
    removeButton.simulate('click');
    expect(onRemoveButtonClick).toHaveBeenCalledTimes(1);
    expect(onRemoveButtonClick).toHaveBeenLastCalledWith('PASS');
  });

  it('calls mover on move button click', () => {
    const onRemoveButtonClick = (): null => null;
    const onMoveButtonClick = jest.fn();

    const output = mount(
      <CityMeta
        onMoveButtonClick={onMoveButtonClick}
        onRemoveButtonClick={onRemoveButtonClick}
        name="PASS"
        distance={10}
      />,
    );
    const moveButton = output.find('button.move');
    expect(onMoveButtonClick).not.toHaveBeenCalled();
    moveButton.simulate('click');
    expect(onMoveButtonClick).toHaveBeenCalledTimes(1);
    expect(onMoveButtonClick).toHaveBeenLastCalledWith('PASS');
  });
});
