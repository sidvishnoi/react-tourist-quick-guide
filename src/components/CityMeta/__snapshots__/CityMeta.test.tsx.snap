// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componets - CityMeta matches snapshot 1`] = `
<div
  className="sc-bwzfXH faEXFt"
>
  <div
    className="buttons"
  >
    <button
      className="move sc-bdVaJa bilwKe"
      color="#fff"
      onClick={[Function]}
      title="Move destination up in list"
    >
      ⬆
    </button>
    <button
      className="remove sc-bdVaJa bxWWPb"
      color="#fff"
      onClick={[Function]}
      title="Remove from list"
    >
      ✗
    </button>
  </div>
  <h2
    className="sc-htpNat dpsiEy"
  >
    PASS
  </h2>
  <div
    className="sc-bxivhb ddgbhp"
  >
    10 KM
  </div>
</div>
`;
